src/i16x16_.rs

```diff
@@ -662,6 +662,130 @@ impl i16x16 {
+          self.arr[1].saturating_add(rhs.arr[1]),
+          self.arr[2].saturating_add(rhs.arr[2]),
+          self.arr[3].saturating_add(rhs.arr[3]),
+          self.arr[4].saturating_add(rhs.arr[4]),
+          self.arr[5].saturating_add(rhs.arr[5]),
+          self.arr[6].saturating_add(rhs.arr[6]),
+          self.arr[7].saturating_add(rhs.arr[7]),
+          self.arr[8].saturating_add(rhs.arr[8]),
+          self.arr[9].saturating_add(rhs.arr[9]),
+          self.arr[10].saturating_add(rhs.arr[10]),
+          self.arr[11].saturating_add(rhs.arr[11]),
+          self.arr[12].saturating_add(rhs.arr[12]),
+          self.arr[13].saturating_add(rhs.arr[13]),
+          self.arr[14].saturating_add(rhs.arr[14]),
+          self.arr[15].saturating_add(rhs.arr[15]),
+        ]}
+      }
+    }
+  }
+  #[inline]
+  #[must_use]
+  pub fn saturating_sub(self, rhs: Self) -> Self {
+    pick! {
+      if #[cfg(target_feature="avx2")] {
+        Self { avx2: sub_saturating_i16_m256i(self.avx2, rhs.avx2) }
+      } else if #[cfg(target_feature="sse2")] {
+        Self { sse0: sub_saturating_i16_m128i(self.sse0, rhs.sse0), sse1: sub_saturating_i16_m128i(self.sse1, rhs.sse1) }
+      } else if #[cfg(target_feature="simd128")] {
+        Self { simd0: i16x8_sub_sat(self.simd0, rhs.simd0), simd1: i16x8_sub_sat(self.simd1, rhs.simd1) }
+      } else {
+        Self { arr: [
+          self.arr[0].saturating_sub(rhs.arr[0]),
+          self.arr[1].saturating_sub(rhs.arr[1]),
+          self.arr[2].saturating_sub(rhs.arr[2]),
+          self.arr[3].saturating_sub(rhs.arr[3]),
+          self.arr[4].saturating_sub(rhs.arr[4]),
+          self.arr[5].saturating_sub(rhs.arr[5]),
+          self.arr[6].saturating_sub(rhs.arr[6]),
+          self.arr[7].saturating_sub(rhs.arr[7]),
+          self.arr[8].saturating_sub(rhs.arr[8]),
+          self.arr[9].saturating_sub(rhs.arr[9]),
+          self.arr[10].saturating_sub(rhs.arr[10]),
+          self.arr[11].saturating_sub(rhs.arr[11]),
+          self.arr[12].saturating_sub(rhs.arr[12]),
+          self.arr[13].saturating_sub(rhs.arr[13]),
+          self.arr[14].saturating_sub(rhs.arr[14]),
+          self.arr[15].saturating_sub(rhs.arr[15]),
+        ]}
+      }
+    }
+  }
+
+  /// Multiply and scale equivilent to ((self * rhs) + 0x4000) >> 15 on each
+  /// lane, effectively multiplying by a 16 bit fixed point number between -1
+  /// and 1. This corresponds to the following instructions:
+  /// - vqrdmulhq_n_s16 instruction on neon
+  /// - i16x8_q15mulr_sat on simd128
+  /// - _mm256_mulhrs_epi16 on avx2
+  /// - emulated via mul_i16_* on sse2
+  #[inline]
+  #[must_use]
+  pub fn mul_scale_round(self, rhs: Self) -> Self {
+    pick! {
+      if #[cfg(target_feature="avx2")] {
+        Self { avx2: mul_i16_scale_round_m256i(self.avx2, rhs.avx2) }
+      } else if #[cfg(target_feature="sse2")] {
+        // unfortunately mul_i16_scale_round_m128i only got added in sse3
```

Comment:
another branch can be added here for sse3 if you'd like.

Reply:
ok... it's for ssse3. not sure why there is also sse3.